cmd/kube-rbac-proxy/app/options/options.go

```diff
@@ -140,17 +148,133 @@ func (o *ProxyRunOptions) Flags() k8sapiflag.NamedFlagSets {
+	if err := flagset.MarkHidden("log-backtrace-at"); err != nil {
+		panic(err)
+	}
+	flagset.StringVar(&o.disabled.logdir, "log-dir", "", "[DISABLED]")
+	if err := flagset.MarkHidden("log-dir"); err != nil {
+		panic(err)
+	}
+	flagset.StringVar(&o.disabled.logfile, "log-file", "", "[DISABLED]")
+	if err := flagset.MarkHidden("log-file"); err != nil {
+		panic(err)
+	}
+	flagset.Uint32Var(&o.disabled.logfilemaxsize, "log-file-max-size", 1800, "[DISABLED]")
+	if err := flagset.MarkHidden("log-file-max-size"); err != nil {
+		panic(err)
+	}
+	flagset.BoolVar(&o.disabled.oneoutput, "one-output", false, "[DISABLED]")
+	if err := flagset.MarkHidden("one-output"); err != nil {
+		panic(err)
+	}
+	flagset.BoolVar(&o.disabled.skipheaders, "skip-headers", false, "[DISABLED]")
+	if err := flagset.MarkHidden("skip-headers"); err != nil {
+		panic(err)
+	}
+	flagset.BoolVar(&o.disabled.skiplogheaders, "skip-log-headers", false, "[DISABLED]")
+	if err := flagset.MarkHidden("skip-log-headers"); err != nil {
+		panic(err)
+	}
+	flagset.IntVar(&o.disabled.stderrthreshold, "stderrthreshold", 2, "[DISABLED]")
+	if err := flagset.MarkHidden("stderrthreshold"); err != nil {
+		panic(err)
+	}
 
 	return namedFlagSets
 }
+
+func (o *ProxyRunOptions) Validate() error {
+	var errs []error
+
+	hasCerts := !(o.TLS.CertFile == "") && !(o.TLS.KeyFile == "")
+	hasInsecureListenAddress := o.InsecureListenAddress != ""
+	if !hasCerts || hasInsecureListenAddress {
+		klog.Warning(`
+==== Deprecation Warning ======================
+
+Insecure listen address will be removed.
+Using --insecure-listen-address won't be possible!
+
+The ability to run kube-rbac-proxy without TLS certificates will be removed.
+Not using --tls-cert-file and --tls-private-key-file won't be possible!
+
+For more information, please go to https://github.com/brancz/kube-rbac-proxy/issues/187
+
+===============================================
+
+		`)
+	}
+
+	if o.TLS.ReloadInterval != time.Minute {
+		klog.Warning(`
+==== Deprecation Warning ======================
+
+tls-reload-interval will be removed.
+Using --tls-reload-interval won't be possible!
+
+For more information, please go to https://github.com/brancz/kube-rbac-proxy/issues/196
+
+===============================================
+		`)
+
+	}
+
+	if len(o.AllowPaths) > 0 && len(o.IgnorePaths) > 0 {
+		errs = append(errs, fmt.Errorf("cannot use --allow-paths and --ignore-paths together"))
+	}
+
+	for _, pathAllowed := range o.AllowPaths {
+		_, err := path.Match(pathAllowed, "")
+		if err != nil {
+			errs = append(errs, fmt.Errorf("failed to verify allow path: %s", pathAllowed))
+		}
+	}
+
+	for _, pathIgnored := range o.IgnorePaths {
+		_, err := path.Match(pathIgnored, "")
+		if err != nil {
+			errs = append(errs, fmt.Errorf("failed to verify ignored path: %s", pathIgnored))
+		}
+	}
+
+	// Removed upstream flags shouldn't be use
+	for _, disabledOpt := range []string{
+		"logtostderr",
```

Comment:
this is a second spot where you're constructing this list, maybe it would make sense to keep it somewhere where you can refer to it